db/migrate/20171222120314_add_constraints_on_discovery_rules_hostgroups.rb

```diff
@@ -0,0 +1,17 @@
+class AddConstraintsOnDiscoveryRulesHostgroups < ActiveRecord::Migration[4.2]
+  def change
+    # turn off Foreign Key checks
```

Comment:
@lzap I feel if we remove the checks, then it might cause a huge issue during `upgrade`. The checks do not affect the success/failure of the migration, the idea here is to be safe during upgrade.
Another option here would be to remove the checks and write code to delete the discovery rules which do not have the appropriate hostgroup which again might cause problems during the upgrade process.

Reply:
I vote deleting them instead, it's not big deal, if no hostgroup was set, they are likely some unused leftovers.